drivers/i2c/i2c_ambiq.c

```diff
@@ -134,21 +197,41 @@ static int i2c_ambiq_init(const struct device *dev)
 
 	data->iom_cfg.eInterfaceMode = AM_HAL_IOM_I2C_MODE;
 
-	ret = am_hal_iom_initialize((config->base - REG_IOM_BASEADDR) / config->size,
-				    &data->IOMHandle);
+	if (AM_HAL_STATUS_SUCCESS !=
+	    am_hal_iom_initialize((config->base - REG_IOM_BASEADDR) / config->size,
+				  &data->IOMHandle)) {
+		LOG_ERR("Fail to initialize I2C\n");
+		return -ENXIO;
+	}
 
-	ret = config->pwr_func();
+	config->pwr_func();
```

Comment:
if pwr_func() is fallible (as it seems to be given the ret = prior to this change, that should likely be accounted for